## Librerias

In [1]:
import pandas as pd
import numpy as np

## Carga de datasets

In [6]:
url = 'https://raw.githubusercontent.com/ronalcabrera/PG_Olist/main/Datasets/olist_sellers_dataset.csv'
sellers = pd.read_csv(url, delimiter=',', encoding='UTF-8')
sellers

seller_id  seller_zip_code_prefix  \
0     3442f8959a84dea7ee197c632cb2df15                   13023   
1     d1b65fc7debc3361ea86b5f14c68d2e2                   13844   
2     ce3ad9de960102d0677a81f5d0bb7b2d                   20031   
3     c0f3eea2e14555b6faeea3dd58c1b1c3                    4195   
4     51a04a8a6bdcb23deccc82b0b80742cf                   12914   
...                                ...                     ...   
3090  98dddbc4601dd4443ca174359b237166                   87111   
3091  f8201cab383e484733266d1906e2fdfa                   88137   
3092  74871d19219c7d518d0090283e03c137                    4650   
3093  e603cf3fec55f8697c9059638d6c8eb5                   96080   
3094  9e25199f6ef7e7c347120ff175652c3b                   12051   

            seller_city seller_state  
0              campinas           SP  
1            mogi guacu           SP  
2        rio de janeiro           RJ  
3             sao paulo           SP  
4     braganca paulista           SP  
...                 ...          ...  
3090            sarandi           PR  
3091            palhoca           SC  
3092          sao paulo           SP  
3093            pelotas           RS  
3094            taubate           SP  

[3095 rows x 4 columns]

## Columnas `seller_id`

Verifico si hay duplicados:

In [7]:
sellers[['seller_id','seller_city']].groupby(['seller_id']).count()

seller_city
seller_id                                    
0015a82c2db000af6aaaf3ae2ecb0532            1
001cca7ae9ae17fb1caed9dfb1094831            1
001e6ad469a905060d959994f1b41e4f            1
002100f778ceb8431b7a1020ff7ab48f            1
003554e2dce176b5555353e4f3555ac8            1
...                                       ...
ffcfefa19b08742c5d315f2791395ee5            1
ffdd9f82b9a447f6f8d4b91554cc7dd3            1
ffeee66ac5d5a62fe688b9d26f83f534            1
fffd5413c0700ac820c7069d66d98c89            1
ffff564a4f9085cd26170f4732393726            1

[3095 rows x 1 columns]

Se verifica que hay la misma cantidad de filas luego de agrupar por `seller_id`, no hay seller_id duplicados.

## Columna `seller_zip_code_prefix`

Verifico el tipo de dato este correcto

In [8]:
sellers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3095 entries, 0 to 3094
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   seller_id               3095 non-null   object
 1   seller_zip_code_prefix  3095 non-null   int64 
 2   seller_city             3095 non-null   object
 3   seller_state            3095 non-null   object
dtypes: int64(1), object(3)
memory usage: 96.8+ KB


Por el tipo de dato de la columna `customer_zip_code_prefix` se verifica que son todos números enteros, procedo a listarlos y contarlos.

In [10]:
zip_code_prefix = sellers['seller_zip_code_prefix'].unique().tolist()
zip_code_prefix.sort()
len(zip_code_prefix)

2246

Verifico que los prefijos y las abreviaturas esten bien

Primero verifico que los valores mínimos esten dentro del rango, luego verifico los máximos

In [11]:
df = sellers[['seller_state','seller_zip_code_prefix']].groupby(['seller_state'], ).min()
df['prefix_max'] = sellers[['seller_state','seller_zip_code_prefix']].groupby(['seller_state'], ).max().values
df.sort_values('seller_zip_code_prefix')

seller_zip_code_prefix  prefix_max
seller_state                                    
SP                              1001       95076
RJ                             20020       28990
RN                             21210       59775
ES                             29010       29704
MG                             30111       39801
BA                             40130       48602
SE                             49055       49980
PE                             50751       55602
PB                             58030       58865
CE                             60025       63540
PI                             64033       64033
MA                             65072       65072
AM                             69005       69005
AC                             69900       69900
DF                             70070       73020
GO                             72801       76500
RO                             76804       76900
MT                             78020       78820
MS                             79090       79400
PR                             80010       87900
PA                             85960       85960
SC                             88010       89900
RS                             90010       99730

Problemas:<br>
• SP presenta un rango que abarca todos los prefijos(1001 hasta 95076)<br>
• RJ y RN comparten rango entre 21210-28990<br>
• ES, MG, BA, SE, PE y PB estan incluídas dentro del rango de RN (desde 29010 hasta 58865)<br>
• PA dentro del rango de PR (85960)<br>
<br>
Creamos tabla maestro de códigos postales reemplazamos por los ID únicos, razón por la cuál no se corregirán estos errores.

## Columna `seller_city`

No vamos a trabajar con esta columna, asi que no se le hará ningún tratamiento

## Columna `seller_state`

In [13]:
print('Cantidad de federaciones:',len(sellers.seller_state.unique().tolist()))
print(sellers.seller_state.unique().tolist())

Cantidad de federaciones: 23
['SP', 'RJ', 'PE', 'PR', 'GO', 'SC', 'BA', 'DF', 'RS', 'MG', 'RN', 'MT', 'CE', 'PB', 'AC', 'ES', 'RO', 'PI', 'MS', 'SE', 'MA', 'AM', 'PA']


Los códigos son correctos, nada que agregar.